### PR TITLE
Feature/l476rg lvgl example

### PR DESCRIPTION
**Pull request by frnktank**
Added a fully functional lvgl example for the l476rg with a few tweaks to the test screen.
Inspired by the example for the l452re.

### CODE REVIEW

File: examples/nucleo_l476rg/lvgl/main.c [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+static constexpr size_t buf_size = 240 * 320 / 8;
+static lv_color_t modm_aligned(4) buf[buf_size];
+
+// Define display buffer 2
+static constexpr size_t buf2_size = 240 * 320 / 8;
+static lv_color_t modm_aligned(4) buf2[buf2_size];
+
+void my_touchpad_read(lv_indev_t*, lv_indev_data_t* data)
+{
+	data->state = RF_CALL_BLOCKING(touchController.isTouched()) ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.c]:
```suggestion
	data->state = touchController.isTouched() ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
```
Resumables are deprecated now

File: examples/nucleo_l476rg/lvgl/main.c [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	if(data->state == LV_INDEV_STATE_PRESSED) {
+		std::tuple xy = RF_CALL_BLOCKING(touchController.getTouchPosition());
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.c]:
```suggestion
		std::tuple xy = touchController.getTouchPosition();
```

File: examples/nucleo_l476rg/lvgl/main.c [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		data->point.x = std::get<0>(xy);
+		data->point.y = std::get<1>(xy);
+
+		std::tuple xyRAW = RF_CALL_BLOCKING(touchController.getRawValues());
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.c]:
```suggestion
		std::tuple xyRAW = touchController.getRawValues();
```

File: examples/nucleo_l476rg/lvgl/main.c [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+main()
+{
+	Board::initialize();
+	Dma1::enable();
+
+	tft::Spi::connect<
+		tft::Sck::Sck,
+		tft::Miso::Miso,
+		tft::Mosi::Mosi>();
+	tft::Spi::initialize<Board::SystemClock, modm::MHz(24)>();
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.c]:
```suggestion
	tft::Spi::initialize<Board::SystemClock, 24_MHz>();
```

File: examples/nucleo_l476rg/lvgl/main.c [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	tftController.initialize();
+	tftController.enableBacklight(true);
+
+	touch::Spi::connect<
+		touch::Sck::Sck,
+		touch::Miso::Miso,
+		touch::Mosi::Mosi>();
+	touch::Spi::initialize<Board::SystemClock, modm::MHz(1.5)>();
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.c]:
```suggestion
	touch::Spi::initialize<Board::SystemClock, 1.5_MHz>();
```

**Review by salkinium — CHANGES_REQUESTED** [review]
Nice!

**Comment by frnktank** [comment]
Removed the resumables and changed to using literals now.

**Comment by frnktank** [comment]
Currently fighting with getting the tests to work

File: examples/nucleo_l476rg/lvgl/main.cpp [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+main()
+{
+	Board::initialize();
+	Dma1::enable();
+
+	tft::Spi::connect<
+		tft::Sck::Sck,
+		tft::Miso::Miso,
+		tft::Mosi::Mosi>();
+	tft::Spi::initialize<Board::SystemClock, 24_Mhz>();
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.cpp]:
```suggestion
	tft::Spi::initialize<Board::SystemClock, 24_MHz>();
```

File: examples/nucleo_l476rg/lvgl/main.cpp [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+	tftController.initialize();
+	tftController.enableBacklight(true);
+
+	touch::Spi::connect<
+		touch::Sck::Sck,
+		touch::Miso::Miso,
+		touch::Mosi::Mosi>();
+	touch::Spi::initialize<Board::SystemClock, 1.5_Mhz>();
```
Comment (salkinium) [inline examples/nucleo_l476rg/lvgl/main.cpp]:
```suggestion
	touch::Spi::initialize<Board::SystemClock, 1.5_MHz>();
```

**Review by salkinium — COMMENTED** [review]
almost there

**Comment by salkinium** [comment]
I'm trying to fix the CI, I might have broken it in some weird way with the upgrade to GCC13 in #1237.

**Review by salkinium — APPROVED** [review]

**Review by salkinium — APPROVED** [review]
Apologies for the many pushes, there were a few issues that popped up all at once on the CI, which had nothing to do with your example. But now it's all fixed, thanks for the contribution!

**Pull request merged**